scripts/build.py

```diff
@@ -552,51 +665,59 @@ def _execute_tasks(tasks, batch_size=24):
 
     Runs no more than 'batch_size' tasks at a time.
     """
-    remaining_tasks = list(tasks)
+    remaining_tasks = collections.deque(tasks)
     currently_running_tasks = []
 
     while remaining_tasks or currently_running_tasks:
         if currently_running_tasks:
-            for task in list(currently_running_tasks):
+            for task in collections.deque(currently_running_tasks):
                 if not task.is_alive():
                     currently_running_tasks.remove(task)
-
         while remaining_tasks and len(currently_running_tasks) < batch_size:
-            task = remaining_tasks.pop()
+            task = remaining_tasks.popleft()
             currently_running_tasks.append(task)
-            task.start()
+            try:
+                task.start()
+            except RuntimeError as threadAlreadyStarted:
+                raise OSError(threadAlreadyStarted.message)
 
 
-def build_files(source, target, file_hashes, file_formats=None):
-    """If no specific file formats is provided, minifies all CSS and JS files,
-    removes whitespace from HTML and interpolates paths in HTML to include
-    hashes in source directory and copies it to target.
-    If specific file formats is provided, only files pertaining to the provided
-    formats will be minified.
+def generate_build_task_to_build_files(
+        source, target, file_hashes, file_formats=None):
+    """If no specific file formats is provided, queue up build task to minify
+    all CSS and JS files, removes whitespace from HTML and interpolates paths
+    in HTML to include hashes in source directory and copies it to target.
+    If specific file formats is provided, only queue up build task to build
+    files pertaining to the provided formats will be minified.
 
     Args:
         source: str. Path relative to /oppia directory of directory
             containing files and directories to be copied.
         target: str. Path relative to /oppia directory of directory where
             to copy the files and directories.
-        file_hashes: dict(str, str). Dictionary of file hashes.
+        file_hashes: dict(str, str). Dictionary with filepaths as keys and
+        hashes of file content as values.
```

Comment:
Indent by 4.

Reply:
Fixed here and elsewhere...I should be more careful!